ludwig/datasets/ohsumed/ohsumed.py

```diff
@@ -0,0 +1,102 @@
+#! /usr/bin/env python
+# coding=utf-8
+# Copyright (c) 2019 Uber Technologies, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import csv
+import os
+import yaml
+import shutil
+from pathlib import Path
+import pandas as pd
+from ludwig.datasets.base_dataset import BaseDataset
+
+"""A class to download, process and return the ohsumed training data to be plugged into the Ludwig training API"""
+
+
+class OhsuMed(BaseDataset):
+
+    def __init__(self):
+        self._initial_path = os.path.abspath(os.path.dirname(__file__))
+        self._config_file_location = os.path.join(self._initial_path, "../text/versions.yaml")
+        self._source_location = os.path.join(self._initial_path, "../text/ohsumed/ohsumed-allcats.csv")
+        self._cache_dir = str(Path.home().joinpath('.ludwig_cache'))
```

Comment:
This stuff we can move into `BaseDataset`.

Reply:
Moved